### ISSUE TITLE
Not an issue: changing det-dize / processing speed

### ISSUE THREAD

**Issue by instant-high** [neuralchen/SimSwap#65]
I've been playing around with the line

_app.prepare(ctx_id= 0, det_thresh=opt.det_thresh, det_size=(640, 640))_

in _test_video_swapsingle.py_

Setting lower values (480, 320, 256, 160) significant increases the processing speed
Setting to 256 doubles  #it/sec. 

It nearly always gives the same results, depending on video resolution and size of the face in the video frame.
eg. if your video is 720p height and the face is someting like 256pixel height in it you can use det-size 256 and it is working.

But arcface throws a lot of warnings for every frame:
 _[W:onnxruntime:, execution_frame.cc:721 onnxruntime::ExecutionFrame::VerifyOutputSizes] Expected shape from model of {12800,10} does not match actual shape of {3200,10} for output 454_

Can this be done a better way?

**Comment by NNNNAI** [comment]
Maybe you can try 320*320. And I thought you can go to insightface to ask the author, cause I just use it as a tool to detect face Hhhh.

**Comment by instant-high** [comment]
Yes. You have to use all other values as a pair (640,640 or 320,320 or.....)
Searched the net so far but couldn't find more information so I asked

**Comment by NNNNAI** [comment]
You can create a issue right here [insightface](https://github.com/deepinsight/insightface), they should be professional enough to solve your problem.

**Comment by rayanarman** [comment]
I tried your suggestion to reduce the face detection size to less than 640, as low as 160. I noticed the significant speed up, but there is an issue i noticed. On multiple faces in the video (side by side VR) it will not detect/swap all of the faces?

**Comment by nttstar** [comment]
You can call `onnxruntime.set_default_logger_severity(3)` to avoid warning.

**Comment by instant-high** [comment]
Thank you.  That works.

**Comment by coldplayer12** [comment]
> 
> 
> I've been playing around with the line
> 
> _app.prepare(ctx_id= 0, det_thresh=opt.det_thresh, det_size=(640, 640))_
> 
> in _test_video_swapsingle.py_
> 
> Setting lower values (480, 320, 256, 160) significant increases the processing speed
> Setting to 256 doubles #it/sec.
> 
> It nearly always gives the same results, depending on video resolution and size of the face in the video frame.
> eg. if your video is 720p height and the face is someting like 256pixel height in it you can use det-size 256 and it is working.
> 
> But arcface throws a lot of warnings for every frame:
> _[W:onnxruntime:, execution_frame.cc:721 onnxruntime::ExecutionFrame::VerifyOutputSizes] Expected shape from model of {12800,10} does not match actual shape of {3200,10} for output 454_
> 
> Can this be done a better way?

Excuse me but, in which file do you find that value? I'm modifying "face_detect_crop_single.py" but it doesn't change the resolution or the det size. Thanks!

**Comment by instant-high** [comment]
As described above.
Change value in test_video_swapsingle.py

**Comment by cwalt2014** [comment]
> As described above. Change value in test_video_swapsingle.py

So I tried changing from 640 to 320, but still only seeing 1 it/s. My graphics card is a Titan V.

**Comment by vladosShikos** [comment]
> Got lost trying to figure out where to call it from, can anybody help?

 I've only managed to find an import of set_default_logger_severity function in onnxruntime\__init__.py, but so far I was unable to pinpoint where are the calls to onnxruntime.
